homeassistant/components/recorder/services.py

```diff
@@ -37,13 +38,14 @@
 
 SERVICE_PURGE_ENTITIES_SCHEMA = vol.Schema(
     {
+        vol.Optional(ATTR_ENTITY_ID, default=[]): cv.entity_ids,
```

Comment:
# Case 1 - No filter
    # - All entities included

If there are no values it will remove everything so maybe we need a one of here?

Reply:
example https://github.com/home-assistant/core/blob/6dcfe861fdc1907cdd88909d4a88ab2de4029586/homeassistant/helpers/config_validation.py#L460

Reply:
Isn't that the intended, documented behavior?

> Leaving all three parameters empty will result in all entities being selected for purging.
> Note: The entity_id is only optional when used in automations.yaml or scripts.yaml. When using the UI to call this service then it is mandatory to specify at least one entity_id using the Target Picker or via YAML mode.

This is kind of an odd case where we partially rely on the frontend to not allow the user to submit this case through the devtools/services, even though it's technically supported 😕

Reply:
I can see how a behavior change for this `purge_entities` servicce might make sense (I don't know why `purge` and `purge_entities` both can target all entities), but my first instinct was not for a breaking change here.

Reply:
I think a breaking change is desirable here to not make mistakes as they can already use the `purge` service to do everything

Reply:
If we make the breaking change to require at least one key, can we drop this behavior as well?

> When using the UI to call this service then it is mandatory to specify at least one entity_id using the Target Picker or via YAML mode.

It seems an odd restriction, that user is disallowed to use the service devtool to purge an entity_glob or domain.

Reply:
> If we make the breaking change to require at least one key, can we drop this behavior as well?
> 
> > When using the UI to call this service then it is mandatory to specify at least one entity_id using the Target Picker or via YAML mode.
> 
> It seems an odd restriction, that user is disallowed to use the service devtool to purge an entity_glob or domain.

I _think_ the answer is yes there.  They should be able to pick any combination of them together to build the filter as long as they pick at least one so it doesn't select everything

Reply:
Hmm I'm seeing that passing `entity_id: []` is both satisfying the `has_at_least_one_key` check, but also still deleting all entities. I think I may want some kind of check that it has at least one key that is not an empty array.